Release v0.5 (#202)
jldec/projectriff.io: website/blog/2020-02-07-announcing-riff-0-5-0.md

---
title: "Announcing riff v0.5.0"
---

> 

Reviewer: things worth calling out that are not discussed:

- kapp is encouraged for installs
- helm is no longer supported for installs
- each runtime (core, knative, streaming) can be installed independently at any time, and uninstalled independently
- Deployers and Processors now consistently use PodTemplateSpec to define containers and volumes
- CRD status now favor object references instead of resource names

> We are happy to announce riff v0.5.0. Thanks to all riff, Cloud Native Buildpack, Knative, Liiklus, Keda, Contour, and other contributors.

Reviewer: ```suggestion
We are happy to announce riff v0.5.0. Thanks to all riff, Cloud Native Buildpack, Knative, Liiklus, KEDA, Contour, and other contributors.
```

The riff CLI can be downloaded from our [releases page](https://github.com/projectriff/cli/releases/tag/v0.5.0) on GitHub. The [getting started](/docs/v0.5/getting-started) guides will help you to run your first function.

![Cambridge UK bridge](assets/cam-bridge.jpg)

## Streaming

> This release introduces a new experimental streaming capability. Functions deployed using the [Streaming Runtime](/docs/v0.5/runtimes/streaming) are able to consume and produce streams of events. The Node and Java function invokers have been extended to support a new [streaming invoker specification](https://github.com/projectriff/invoker-specification/blob/master/streaming.md).

Reviewer: we should talk a bit about how this is different than what riff has had in the past

Author: The text mentions multiple input streams and gateways. I'd rather leave it like that while we're still evolving rather than drawing comparisons with specific designs in the past.

Here is an example of a JavaScript streaming function interface with 2 input and 1 output stream, using the [node function invoker]().
```js

> module.exports = (inputStreams, outputStreams) => {

Reviewer: if you want to be more fancy:

```js
module.exports = ({ impressions, orders }, { conversions }) => {
     // do something
 };
 module.exports.$interactionModel = 'node-streams';
```

Author: That's cool - i think we should stick with the slightly longer version for clarity.

    const { impressions, orders } = inputStreams;
    const { conversions } = outputStreams;
    // do something
};
module.exports.$interactionModel = 'node-streams';
```
<!--truncate-->

#### Gateways and Processors

> [Gateways](docs/v0.5/cli/riff-streaming-kafka-gateway) connect streams to a messaging service like Kafka. [Processors](/docs/v0.5/cli/riff-streaming-processor-create) connect functions with gateways.

Reviewer: ```suggestion
[Gateways](docs/v0.5/cli/riff-streaming-gateway) connect streams to a messaging service like Kafka or Pulsar. [Processors](/docs/v0.5/cli/riff-streaming-processor-create) connect functions with gateways.
```

> ![Streaming architecture diagram](assets/streaming.png)

Reviewer: we should use a diagram that explains how 0.5 is implemented, not what we're thinking about for 0.6

> This release includes Gateways for Kafka and Pulsar, and a lightweight in-memory gateway. All three are based on [Liiklus](https://github.com/bsideup/liiklus).

Reviewer: ```suggestion
This release includes Gateways for Kafka and Pulsar, and a non-production in-memory gateway. All three are based on [Liiklus](https://github.com/bsideup/liiklus).
```

#### Streaming Next Steps

> We are discussing this streaming model with the Knative community, and exploring ways to enable streaming for other types of workloads beyond streaming functions.

Reviewer: The discussions are broader than the knative community
```suggestion
We are discussing this streaming model with the community, and exploring ways to enable streaming for other types of workloads beyond streaming functions.
```

## Other Highlights of v0.5.0

### Knative Runtime

- The Knative runtime has been upgraded to use Knative Serving v0.12.1.

> - Istio has been replaced with Contour as the ingress, and works with both the Knative and the Core runtime. Note that `--ingress-policy External` is now [required](/docs/v0.5/cli/riff-knative-deployer-create) in order to route HTTP traffic from outside the cluster.

Reviewer: ```suggestion
- [Contour](https://projectcontour.io) has replaced Istio as the ingress controller, and works with both the Knative and the Core runtime. Note that Deployers are only accessible from inside the cluster by default, `--ingress-policy External` is now [required](/docs/v0.5/cli/riff-knative-deployer-create) in order to route HTTP traffic from outside the cluster.
```

- The CLI also provides for configuration of
  -  `--min-scale` and `--max-scale`
  -  `--container-concurrency`

> ### Core Runtime

Reviewer: should we call out that we're considering deprecating the core runtime?

Author: I'd prefer to make a more general statement once we have a clearer picture of the riff + Knative plan.

> - We have added Ingress support to the Core Runtime, making it easier direct HTTP requests to functions, apps, and containers deployed using the Core runtime.

Reviewer: ```suggestion
- We added Ingress support to the Core Runtime, allowing external HTTP requests to target functions, apps, and containers.
```

### Build

> - Container images managed by `riff container create` are now resolved into image digests, and the controller polls the registry to detect image updates and propagate the changes to deployed workloads.

Reviewer: this isn't new, it existed in 0.4

Author: This was based on https://github.com/projectriff/system/commit/c69abe9d3c9c526b744d77f1392ba8b6b533d911.

I'll remove the section for now, or someone can try to reword.
cc. @ekcasey
